File: src/AnalyticsService/Transport/Validation/BatchPeriodStatRequestValidator.fs
namespace AnalyticsService.Transport.Validation

open AnalyticsService.Transport.Contracts.Requests
open FluentValidation

/// Validator class for BatchPeriodStatRequest record.
[<Sealed>]
type internal BatchPeriodStatRequestValidator() as this =
    inherit AbstractValidator<BatchPeriodStatRequest>()
    do
        this.RuleFor(fun i -> i.WorkflowId).NotEmpty() |> ignore
        this.RuleFor(fun i -> i.StartDate).NotEmpty() |> ignore
        this
            .RuleFor(fun i -> i.EndDate)
            .NotEmpty()
            .GreaterThan(fun i -> i.StartDate) |> ignore

File: src/AnalyticsService/Transport/Validation/BatchPeriodStatRequestValidator.cs
using AnalyticsService.Transport.Contracts.Requests;
using FluentValidation;

namespace AnalyticsService.Transport.Validation;

/// <summary>
/// Validator class for <see cref="BatchPeriodStatRequest"/> record.
/// </summary>
internal sealed class BatchPeriodStatRequestValidator : AbstractValidator<BatchPeriodStatRequest>
{
    public BatchPeriodStatRequestValidator()
    {
        RuleFor(i => i.WorkflowId).NotEmpty();
        RuleFor(i => i.StartDate).NotEmpty();
        RuleFor(i => i.EndDate).NotEmpty().GreaterThan(i => i.StartDate);
    }
}